test/integration/scheduler/scheduler_test.go

```diff
@@ -338,4 +368,168 @@ var _ = ginkgo.Describe("Scheduler", func() {
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		gomega.Expect(createdJob1.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+
+		job2 := testing.MakeJob("on-demand-job2", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "8").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job2)).Should(gomega.Succeed())
+		createdJob2 := &batchv1.Job{}
+		gomega.Consistently(func() bool {
+			lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
+			return k8sClient.Get(ctx, lookupKey, createdJob2) == nil && *createdJob2.Spec.Suspend
+		}, framework.ConsistentDuration, framework.Interval).Should(gomega.BeTrue())
+
+		job3 := testing.MakeJob("on-demand-job3", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job3)).Should(gomega.Succeed())
+		createdJob3 := &batchv1.Job{}
+		gomega.Eventually(func() bool {
+			lookupKey := types.NamespacedName{Name: job3.Name, Namespace: job3.Namespace}
+			return k8sClient.Get(ctx, lookupKey, createdJob3) == nil && !*createdJob3.Spec.Suspend
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		gomega.Expect(createdJob3.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+
+		ginkgo.By("delete job1")
+		gomega.Expect(k8sClient.Delete(ctx, job1)).Should(gomega.Succeed())
+		// Simulated deletion of workload by ownerReference of job1.
+		gomega.Expect(k8sClient.Delete(ctx, workload1)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			wl := &kueue.QueuedWorkload{}
+			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
+			if err := k8sClient.Get(ctx, lookupKey, wl); err != nil && apierrors.IsNotFound(err) {
+				return true
+			}
+			return false
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+
+		gomega.Eventually(func() bool {
+			lookupKey := types.NamespacedName{Name: job2.Name,
+				Namespace: job2.Namespace}
+			return k8sClient.Get(ctx, lookupKey,
+				createdJob2) == nil && !*createdJob2.Spec.Suspend
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		gomega.Expect(createdJob2.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+	})
+
+	ginkgo.It("Should re-enqueue by the delete event of workload belonging to the same Cohort", func() {
+		job1 := testing.MakeJob("on-demand-job1", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "8").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job1)).Should(gomega.Succeed())
+		workload1 := &kueue.QueuedWorkload{}
+		gomega.Eventually(func() bool {
+			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
+			err := k8sClient.Get(ctx, lookupKey, workload1)
+			return err == nil
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
```

Comment:
ditto, no need for this eventually block

Reply:
deleted